### PR TITLE
[Bug]fix for missing id in old nav

### PR DESCRIPTION
**Pull request by sumukhswamy**
### Description
Side tree was not loading in the case of ols nav as the MDS id was not being passed, 
image
after Fix
image

 
### Issues Resolved
There was no ID added from this pr which was causing the page to not refresh
https://github.com/opensearch-project/dashboards-query-workbench/pull/349

 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/Main/main.tsx [hunk 1]
```diff
@@ -1091,7 +1125,11 @@ export class Main extends React.Component<MainProps, MainState> {
                         onSelect={this.handleDataSelect}
                         urlDataSource={this.props.urlDataSource}
                         asyncLoading={this.state.asyncLoading}
-                        dataSourceMDSId={this.props.dataSourceMDSId}
+                        dataSourceMDSId={
+                          coreRefs?.chrome?.navGroup.getNavGroupEnabled()
```
Comment (joshuali925) [inline public/components/Main/main.tsx:1129]: should this be `this.state.newNavEnabled`

File: public/components/Main/main.tsx [hunk 2]
```diff
@@ -295,6 +296,7 @@ export class Main extends React.Component<MainProps, MainState> {
       selectedMDSDataConnectionId: this.props.dataSourceMDSId,
       mdsClusterName: '',
       flintDataConnections: false,
+      newNavEnabled: coreRefs?.chrome?.navGroup.getNavGroupEnabled(),
```
Comment (joshuali925) [inline public/components/Main/main.tsx:299]: this doesn't seem to change, is it necessary to store as a state?

File: public/components/Main/main.tsx [hunk 3]
```diff
@@ -1110,7 +1148,11 @@ export class Main extends React.Component<MainProps, MainState> {
                     updateSQLQueries={this.updateSQLQueries}
                     refreshTree={this.state.refreshTree}
                     dataSourceEnabled={this.props.dataSourceEnabled}
-                    dataSourceMDSId={this.props.dataSourceMDSId}
+                    dataSourceMDSId={
+                      coreRefs?.chrome?.navGroup.getNavGroupEnabled()
+                        ? this.props.dataSourceMDSId
+                        : this.state.selectedMDSDataConnectionId
```
Comment (joshuali925) [inline public/components/Main/main.tsx:1154]: there are many similar checks in this file, can this logic be separated out into a variable?

File: public/components/Main/main.tsx (hunk 2 above)
Comment (sumukhswamy) [reply public/components/Main/main.tsx:299]: the state can change if its toggeled in the settings

**Review by ps48 — APPROVED** [review]

**Review by TackAdam — APPROVED** [review]

**Review by derek-ho — APPROVED** [review]

**Pull request merged**